### PR TITLE
[NPM] Fix ebpf map errors

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

`tcp_ongoing_connect_pid` map filling up leads to `E2BIG` errors from `bpf_map_update_elem` call. This map filling up means that we are not removing entries as often as we should be in `tcp_close` and `tcp_finish_connect` kprobes. The missing case here is if `tcp_connect` returns an error. This could lead to a `tcp_finish_connect` call not happening, or the caller not calling close on the socket, in which case a `tcp_close` will not happen either. The latter could happen if the connect is retried, for example.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/co-re/tracer-fentry.c [hunk 1]
```diff
@@ -457,8 +457,24 @@ int BPF_PROG(tcp_connect, struct sock *sk) {
...
+    // from tcp_ongoing_connect_pid as tcp_finish_connect
+    // kprobe or tcp_close kprobe may not be called
+    // for this tcp_connect call
+    bpf_map_delete_elem(&tcp_ongoing_connect_pid, &sk);
+    return 0;
+}
+
 SEC("fentry/tcp_finish_connect")
-int BPF_PROG(tcp_finish_connect, struct sock *sk, struct sk_buff *skb, int rc) {
+int BPF_PROG(tcp_finish_connect, struct sock *sk, struct sk_buff *skb) {
```
Comment (leeavital) [inline pkg/network/ebpf/c/co-re/tracer-fentry.c:477]: do we strictly need the tcp_finish_connect  if we're hooking the exit of tcp_connect ?

File: pkg/network/ebpf/c/tracer.c [hunk 2]
```diff
@@ -860,10 +860,37 @@ int kprobe__tcp_connect(struct pt_regs *ctx) {
     struct sock *skp = (struct sock *)PT_REGS_PARM1(ctx);
 
     bpf_map_update_with_telemetry(tcp_ongoing_connect_pid, &skp, &pid_tgid, BPF_ANY);
+    bpf_map_update_with_telemetry(tcp_connect_args, &pid_tgid, &skp, BPF_ANY);
 
     return 0;
 }
 
+SEC("kretprobe/tcp_connect")
+int kretprobe__tcp_connect(struct pt_regs *ctx) {
```
Comment (leeavital) [inline pkg/network/ebpf/c/tracer.c:869]: same question as above about tcp_finish_connect

File: pkg/network/ebpf/c/co-re/tracer-fentry.c (hunk 1 above)
Comment (hmahmood) [reply pkg/network/ebpf/c/co-re/tracer-fentry.c:477]: `tcp_connect` just sends the `SYN` off, and gets the ball rolling for the connect. So the connect is not done when it returns. We're just checking now if the `tcp_connect` failed immediately, which is another case where `tcp_finish_connect` will not be called.

Comment (leeavital) [reply pkg/network/ebpf/c/co-re/tracer-fentry.c:477]: Makes sense, thank you!

**Review by leeavital — APPROVED** [review]

**Pull request merged**